builtin/providers/aws/resource_aws_autoscaling_group.go

```diff
@@ -304,6 +312,8 @@ func resourceAwsAutoscalingGroupUpdate(d *schema.ResourceData, meta interface{})
 		AutoScalingGroupName: aws.String(d.Id()),
 	}
 
+	opts.NewInstancesProtectedFromScaleIn = aws.Bool(d.Get("protect_from_scale_in").(bool))
```

Comment:
I attempted to use the following:

```go
if d.HasChange("protect_from_scale_in") {
  opts.NewInstancesProtectedFromScaleIn = aws.Bool(d.Get("protect_from_scale_in").(bool))
}
```

However, I found that the state was never updated. My expectation is that `HasChange` should be used, not yet clear why I was unable to complete updates when using `HasChange`.